env/fs_posix.cc

```diff
@@ -930,6 +931,26 @@ class PosixFileSystem : public FileSystem {
     optimized.fallocate_with_keep_size = true;
     return optimized;
   }
+
+  FileOptions OptimizeForCompactionTableRead(
+      const FileOptions& file_options,
+      const ImmutableDBOptions& db_options) const override {
```

Comment:
I dropped `optimized_file_options.use_direct_reads = db_options.use_direct_reads;` as in file_system.cc https://github.com/facebook/rocksdb/blob/3495c94761629404f5476f732e2130134cfcc51b/env/file_system.cc#L170. 

This is because file_options should already be populated with use_direct_reads in db options before passing in this function.